app/src/main/java/ch/epfl/sweng/rps/ui/game/GameFragment.kt

```diff
@@ -35,11 +35,27 @@ class GameFragment : Fragment() {
     }
 
     private fun rpsPressed(hand: Hand) {
-        matchViewModel.playHand(hand) {
-            findNavController().navigate(R.id.action_gameFragment_to_gameResultFragment)
+        matchViewModel.playHand(hand,
+            updateUIResultCallback = {
+                selectComputerChoice(matchViewModel.gameService?.currentRound?.hands?.get(
+                matchViewModel.computerPlayer!!.computerPlayerId)!!)},
+            navigationCallback = {
+            findNavController().navigate(R.id.action_gameFragment_to_gameResultFragment)}
+        )
+    }
+    private fun selectComputerChoice(hand: Hand){
+        when (hand) {
+            Hand.ROCK -> {
+                binding.rockRBOpponent.isChecked = true
+            }
+            Hand.PAPER -> {
+                binding.paperRBOpponent.isChecked = true
+            }
+            else -> {
+                binding.scissorsRB.isChecked = true
+            }
```

Comment:
This should be exhaustive and not use else.

Reply:
Thanks, you're right. I've changed it now.